Activity 03

Droxinbox opened the pull request:
The Activity 03 is now completed and ready for review.

Droxinbox committed:
Activity 03

The Activity 03 is now completed and ready for review.

erobledoapex commented:
Las instrucciones también piden una captura de pantalla de los resultados en la descripción de la PR

erobledoapex commented on ExercisePart1/src/app/app.component.html:
```diff
@@ -1,336 +1,19 @@
+  <div class="container mt-3">    
+    <label for="" class="form-label">Ingrese texto: </label>
+    <input type="text" class="form-control" [(ngModel)]="textValue">
+    <p class="mt-3" Capitalize>{{textValue.toLowerCase()}}</p>
```
las directivas por convención son escritas en el componente .ts como camelCase y en el html como snake-case

erobledoapex replied:
no es buena práctica usar funciones en el HTML a menos a que ejecute un evento como un clic, el toLowerCase aquí no hay que ponerlo, la instrucción del ejercicio no pide un input, con una simple variable con un texto todo en minúsculas es suficiente.

erobledoapex commented on ExercisePart1/src/app/app.component.ts:
```diff
@@ -5,6 +5,8 @@ import { Component } from '@angular/core';
 })
+
 export class AppComponent {
   title = 'ExercisePart1';
```
las variables deben llevar su tipo, en este caso "string"

erobledoapex commented on ExercisePart1/src/app/directives/capitalize.directive.ts:
```diff
@@ -0,0 +1,13 @@
+export class CapitalizeDirective {
+
+    constructor (private variable: ElementRef) {
+        variable.nativeElement.style.textTransform = 'capitalize';
```
el problema de hacerlo de esta manera es que si obtienes el valor del elemento no viene capitalizado, solo es una transformación en los estilos, favor de hacer la transformación de los datos

Droxinbox committed:
Activity 03 ver2.0

+Screenshot
+camelCase and snake-case
+Input deleted
+Type of variables
+titlecase added

Droxinbox commented:
![Activity 03](https://github.com/Droxinbox/AngularPractices/assets/36482757/52aa9fc3-1461-4baf-8411-8e47d3cc6f39)

Droxinbox committed:
Activity 03 ver2.1

+Activity has been converted to component
+Unnecessary lines of code removed
